scripts/sphinx_extensions/warnings_filter.py

```diff
@@ -0,0 +1,97 @@
+"""
```

Comment:
Why can't we use the upstream one directly?

Reply:
if we want to be self-contained w.r.t our docset extensions it's better to have our own set, I'm fine using the upstream one though.

Reply:
using upstream for now